### PR TITLE
fix issues in course analysis statistics tool

### PR DESCRIPTION
**Pull request by karlandindrakryggen**
The main changes that addresses reported issue:
- faulty week logic for determining semester is removed and only now semester from Kopps is used.
- only one semester can be selected, to avoid issues of added same course offering multiple time.
- ST as selection is removed, course offerings will be included in HT och VT depending on semester from Kopps (ST was caluclated from weeks before and is no actual semester)

The PR aslo include some other fixes for different warnings:
- replaced rendering of `<StatisticsAlert>` through `document.getElementById()` and `createRoot()` to by moving useStatisticsAsync one step up in react tree and rendering it the normal React way
- removed used of deprecated defaultProps
- some style fixes in chart and tables

### CODE REVIEW

File: server/apiCalls/transformers/offerings.js [hunk 1]
```diff
@@ -150,57 +132,47 @@ function _parseTermSeasonForNthWeek(nthWeek) {
...
  * @returns {[]}           Array, containing offerings’ relevant data
  */
-function filterOfferingsForAnalysis(
-  courses = [],
-  chosenSemesters = [],
-  chosenSeasons = [],
-  chosenSchool = '',
-  language = 'sv'
-) {
+function filterOfferingsForAnalysis(courses = [], chosenSemester, chosenSchool = '', language = 'sv') {
```
Comment (karlandindrakryggen) [inline server/apiCalls/transformers/offerings.js:140]: Most notable changes are in this function.

**Comment by karlandindrakryggen** [comment]
URL to the page: 
https://www-r.referens.sys.kth.se/student/kurser/kurs/statistik
http://localhost:3000/student/kurser/kurs/statistik

**Review by axelbjo — APPROVED** [review]
Tested on my machine and it works fine! Nice job with the refactoring the logic surrounding seasons/semesters, it looks a lot less convoluted and unclear now.

**Review by allazis — APPROVED** [review]
LGTM!

Ran locally and clicked around. A lot cleaner now, having the semesters corresponding to KOPPS semesters, and the new table styling that makes it possible to actually see the end date right away.

Good job!

**Review by amirhossein-haerian — APPROVED** [review]
Well done ⭐️
I also tested locally and everything looks great.

**Pull request merged**